libshaderc_util/src/spirv_tools_wrapper.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2016 The Shaderc Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "libshaderc_util/spirv_tools_wrapper.h"
+
+#include <cassert>
+#include <sstream>
+
+#include "spirv-tools/libspirv.h"
+
+namespace {
+
+// Writes the message contained in the diagnostic parameter to *dest. Assumes
+// the diagnostic message is reported for a binary location.
+void SpirvToolsOutputDiagnostic(spv_diagnostic diagnostic, std::string* dest) {
+  std::ostringstream os;
+  if (diagnostic->isTextSource) {
+    os << diagnostic->position.line + 1 << ":"
+       << diagnostic->position.column + 1;
+  } else {
+    os << diagnostic->position.index;
+  }
+  os << ": " << diagnostic->error;
+  *dest = os.str();
+}
+
+}  // anonymous namespace
+
+namespace shaderc_util {
+
+bool SpirvToolsDisassemble(const std::vector<uint32_t>& binary,
+                           std::string* text_or_error) {
+  auto spvtools_context = spvContextCreate(SPV_ENV_UNIVERSAL_1_0);
```

Comment:
Might be better to use `SPV_ENV_VULKAN*` here.  Right now it doesn't matter, but it would be more future-proof.